### DISCUSSION TITLE
FINN Hardware Resource Usage in Pynq-Z2

### DISCUSSION THREAD

**Discussion by Ba1tu3han** [Xilinx/finn#1003] [Q&A] [answered]
Hello,

I have quantized and trained the CNV model which is from Brevitas Github repo. Then, I compiled it by FINN by following CNV end-to-end example and run it in Pynq-Z2.

I would like to learn how much hardware resource of just the FPGA it uses. Do you recommend a guide or a notebook?

Thank you,

Bests,
Batuhan

**Comment by fpjentzsch** [comment, marked as answer]
Hi,

you can inspect the design manually using the Vivado GUI or use the `post_synth_res` FINN analysis pass: https://github.com/Xilinx/finn/blob/766e9fb47f0c31e82c45cfb9384d1ced8b1ddf7a/src/finn/analysis/fpgadataflow/post_synth_res.py#L38

If you use the FINN builder, it will also copy the Vivado utilization report to "report/post_synth_resources.xml" in your output directory.

**Reply by Ba1tu3han** [reply]
Hello Felix,

Thank you for your reply. 

I will use both of them. Then I will decide to which one is more suitable for my case.

 Could you provide me a starting point for Vivado GUI for hardware utilization?

**Reply by auphelia** [reply]
Hi @Ba1tu3han,

Please refer for this to the official documentation about Vivado, e.g. here: https://docs.amd.com/r/en-US/ug906-vivado-design-analysis/Report-Utilization
There are also useful videos on how to navigate in the Vivado GUI that you can find online.
